src/rebuild/initiator.c

```diff
@@ -568,10 +573,22 @@ punch_iod_pack(struct rebuild_one *rdone, daos_iod_t *iod)
 			return -DER_NOMEM;
 	}
 
+	for (i = 0; i < iod->iod_nr; i++) {
+		if (rdone->ro_punch_epoch == 0 ||
+		    iod->iod_eprs[i].epr_lo > rdone->ro_punch_epoch)
+			rdone->ro_punch_epoch = iod->iod_eprs[i].epr_lo;
+	}
```

Comment:
I don't quite follow this, the code means even invisible punch records will be returned?

Reply:
yes, only for the last  one, since we need rebuild punched record.